logo.js

```diff
@@ -546,6 +552,32 @@ function LogoInterpreter(turtle, stream, savehook)
     return lhs;
   }
 
+  function defer(func) {
+    var input = Array.prototype.slice.call(arguments, 1);
+    return function () {
```

Comment:
Each item is a function that returns a promise sometimes (concrete expressions like `1` don't return promises), which makes it yet a bit more complicated, and doesn't feel very reusable.